packages/eslint-plugin/src/rules/naming-convention.ts

```diff
@@ -395,222 +417,244 @@ export default util.createRule<Options, MessageIds>({
+          const scope = context.getScope();
 
-        if (isExported(node, node.id.name, scope)) {
-          modifiers.add(Modifiers.exported);
-        }
+          if (isExported(node, node.id.name, scope)) {
+            modifiers.add(Modifiers.exported);
+          }
 
-        if (isUnused(node.id.name, scope)) {
-          modifiers.add(Modifiers.unused);
-        }
+          if (isUnused(node.id.name, scope)) {
+            modifiers.add(Modifiers.unused);
+          }
 
-        validator(node.id, modifiers);
+          validator(node.id, modifiers);
+        },
       },
 
       // #endregion typeAlias
 
       // #region enum
 
-      TSEnumDeclaration(node): void {
-        const validator = validators.enum;
-        if (!validator) {
-          return;
-        }
-
-        const modifiers = new Set<Modifiers>();
-        // enums create their own nested scope
-        const scope = context.getScope().upper;
+      TSEnumDeclaration: {
+        validator: validators.enum,
+        handler: (node, validator): void => {
+          const modifiers = new Set<Modifiers>();
+          // enums create their own nested scope
+          const scope = context.getScope().upper;
 
-        if (isExported(node, node.id.name, scope)) {
-          modifiers.add(Modifiers.exported);
-        }
+          if (isExported(node, node.id.name, scope)) {
+            modifiers.add(Modifiers.exported);
+          }
 
-        if (isUnused(node.id.name, scope)) {
-          modifiers.add(Modifiers.unused);
-        }
+          if (isUnused(node.id.name, scope)) {
+            modifiers.add(Modifiers.unused);
+          }
 
-        validator(node.id, modifiers);
+          validator(node.id, modifiers);
+        },
       },
 
       // #endregion enum
 
       // #region typeParameter
 
-      'TSTypeParameterDeclaration > TSTypeParameter'(
-        node: TSESTree.TSTypeParameter,
-      ): void {
-        const validator = validators.typeParameter;
-        if (!validator) {
-          return;
-        }
+      'TSTypeParameterDeclaration > TSTypeParameter': {
+        validator: validators.typeParameter,
+        handler: (node: TSESTree.TSTypeParameter, validator): void => {
+          const modifiers = new Set<Modifiers>();
+          const scope = context.getScope();
 
-        const modifiers = new Set<Modifiers>();
-        const scope = context.getScope();
-
-        if (isUnused(node.name.name, scope)) {
-          modifiers.add(Modifiers.unused);
-        }
+          if (isUnused(node.name.name, scope)) {
+            modifiers.add(Modifiers.unused);
+          }
 
-        validator(node.name, modifiers);
+          validator(node.name, modifiers);
+        },
       },
 
       // #endregion typeParameter
     };
+
+    return Object.fromEntries(
+      Object.entries(selectors)
+        .map(([selector, { validator, handler }]) => {
+          if (validator == null) {
+            return null;
+          }
```

Comment:
Precisely 15,213 test cases, and none with a nullable `validator`!

```suggestion
```

In `packages/eslint-plugin/src/rules/naming-convention-utils/types.ts`:

```diff
type ParsedOptions = Record<SelectorsString, null | ValidatorFunction>;
type ParsedOptions = Record<SelectorsString, ValidatorFunction>;
```